src/core/Microsoft.PowerToys.Settings.UI/Views/KeyboardManagerPage.xaml.cs

```diff
@@ -12,12 +20,69 @@ namespace Microsoft.PowerToys.Settings.UI.Views
+        private const string PowerToyName = "Keyboard Manager";
+
+        private readonly CoreDispatcher dispatcher;
+        private readonly FileSystemWatcher watcher;
+
+        public KeyboardManagerViewModel ViewModel { get; }
 
         public KeyboardManagerPage()
         {
+            dispatcher = Window.Current.Dispatcher;
+
+            ViewModel = new KeyboardManagerViewModel(ShellPage.SendDefaultIPCMessage, FilterRemapKeysList);
+
+            watcher = Helper.GetFileWatcher(
+                PowerToyName,
+                ViewModel.settings.Properties.ActiveConfiguration.Value + ".json",
+                OnConfigFileUpdate);
+
             InitializeComponent();
             DataContext = ViewModel;
         }
+
+        private async void OnConfigFileUpdate()
+        {
+            // Note: FileSystemWatcher raise notification multiple times for single update operation.
+            // Todo: Handle duplicate events either by somehow suppress them or re-read the configuration everytime since we will be updating the UI only if something is changed.
+            if (ViewModel.LoadProfile())
+            {
+                await dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+                {
+                    ViewModel.NotifyFileChaged();
+                });
+            }
+        }
+
+        private void CombineRemappings(List<KeysDataModel> remapKeysList, uint leftKey, uint rightKey, uint combinedKey)
+        {
+            KeysDataModel firstRemap = remapKeysList.Find(x => uint.Parse(x.OriginalKeys) == leftKey);
+            KeysDataModel secondRemap = remapKeysList.Find(x => uint.Parse(x.OriginalKeys) == rightKey);
+            if (firstRemap != null && secondRemap != null)
+            {
+                if (firstRemap.NewRemapKeys == secondRemap.NewRemapKeys)
+                {
+                    KeysDataModel combinedRemap = new KeysDataModel
+                    {
+                        OriginalKeys = combinedKey.ToString(),
+                        NewRemapKeys = firstRemap.NewRemapKeys,
+                    };
+                    remapKeysList.Insert(remapKeysList.IndexOf(firstRemap), combinedRemap);
+                    remapKeysList.Remove(firstRemap);
+                    remapKeysList.Remove(secondRemap);
+                }
+            }
+        }
+
+        private int FilterRemapKeysList(List<KeysDataModel> remapKeysList)
+        {
+            CombineRemappings(remapKeysList, (uint)VirtualKey.LeftControl, (uint)VirtualKey.RightControl, (uint)VirtualKey.Control);
+            CombineRemappings(remapKeysList, (uint)VirtualKey.LeftMenu, (uint)VirtualKey.RightMenu, (uint)VirtualKey.Menu);
+            CombineRemappings(remapKeysList, (uint)VirtualKey.LeftShift, (uint)VirtualKey.RightShift, (uint)VirtualKey.Shift);
+            CombineRemappings(remapKeysList, (uint)VirtualKey.LeftWindows, (uint)VirtualKey.RightWindows, Helper.VirtualKeyWindows);
+
+            return 0;
+        }

```

Comment:
I think we should move these two methods to the view model as well, since they don't require any arguments from the UI.

Reply:
@arjunbalgovind the library that has VirtualKey.LeftShift is not compatible with a done standard project. Thats why I moved it.

Reply:
It looks like VirtualKey is there only in UWP. Can you replace it with `Windows.Forms.Keys`? https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.keys?view=netcore-3.1
LShiftKey, RShiftKey, ShiftKey, ControlKey, RControlKey, LControlKey, Menu, LMenu, RMenu, LWin, RWin?

Reply:
okay i will try that too.

Reply:
@arjunbalgovind Seems like Windows.Forms.Keys are not compatible for .Net Standard, is there any reason you feel strongly about moving the methods back to the ViewModel? I also noticed that no tests were written for them and the primary focus of this PR is to get the tests running again with minimal refactoring.

Reply:
Okay, I think we can get it in as is, and the refactoring could be done later.